flashlight/pkg/runtime/common/SequentialBuilder.cpp

```diff
@@ -64,17 +64,20 @@ fl::Variable forwardSequentialModuleWithPadMask(
     std::shared_ptr<fl::Module> ntwrk,
     const af::array& inputSizes) {
   // expected input dims T x C x 1 x B
-  int T = input.dims(0), B = input.dims(3);
+  int B = input.dims(3);
   auto inputMaxSize = af::tile(af::max(inputSizes), 1, B);
-  af::array inputNotPaddedSize = af::ceil(inputSizes * T / inputMaxSize);
-  auto padMask = af::iota(af::dim4(T, 1), af::dim4(1, B)) <
-      af::tile(inputNotPaddedSize, T, 1);
+  af::array inputNotPaddedSizeRatio = inputSizes / inputMaxSize;
   auto ntwrkSeq = std::dynamic_pointer_cast<fl::Sequential>(ntwrk);
   auto output = input;
   for (auto& module : ntwrkSeq->modules()) {
     auto tr = std::dynamic_pointer_cast<fl::Transformer>(module);
     auto cfr = std::dynamic_pointer_cast<fl::Conformer>(module);
     if (tr != nullptr || cfr != nullptr) {
+      /* input dims of Transformer module should be CxTxBx1 */
+      int T = output.dims(1);
```

Comment:
As mentioned -- this is a big assumption that's architecture and layer dependent. Better to resize the padding mask per layer to fit whatever layers are in the architecture or add some abstraction to automatically do this if interested.